src/utils/polymer.dom.html

```diff
@@ -244,7 +244,7 @@
       let list = [];
       for (let i=0, l=c$.length, c; (i<l) && (c=c$[i]); i++) {
         if ((c.nodeType === Node.ELEMENT_NODE) &&
-            Polymer.Utils.matchesSelector(this.node, selector)) {
+            Polymer.Utils.matchesSelector(c, selector)) {
```

Comment:
If you're going down the same rabbit hole as me right now, I think https://github.com/Polymer/polymer/blob/fd052da7f8e4aa7432a9ed36e55345d77e2aa0f3/src/utils/polymer.dom.html#L24 is going to end up being your next change. :) `node.childNodes` -> `(node.root ? node.root : node).childNodes` or something similar.